system/lib/pthread/pthread_create.c

```diff
@@ -68,6 +68,31 @@ static int dummy_getpid(void) {
 }
 weak_alias(dummy_getpid, __syscall_getpid);
 
+static int tl_lock_count;
+static int tl_lock_waiters;
+
+volatile int __thread_list_lock;
+
+void __tl_lock(void) {
+  int tid = __pthread_self()->tid;
+  int val = __thread_list_lock;
+  if (val == tid) {
+    tl_lock_count++;
+    return;
+  }
+  while ((val = a_cas(&__thread_list_lock, 0, tid)))
+    __wait(&__thread_list_lock, &tl_lock_waiters, val, 0);
+}
+
+void __tl_unlock(void) {
+  if (tl_lock_count) {
+    tl_lock_count--;
+    return;
+  }
+  a_store(&__thread_list_lock, 0);
+  if (tl_lock_waiters) __wake(&__thread_list_lock, 1, 0);
+}
+
```

Comment:
Should the `__tl_sync` function also be implemented here?
https://github.com/emscripten-core/emscripten/blob/8647224628dfc10dec6e5c6f3621d625a5f16eda/system/lib/libc/musl/src/thread/pthread_create.c#L45-L52

`pthread_join` seems to use this.
https://github.com/emscripten-core/emscripten/blob/8647224628dfc10dec6e5c6f3621d625a5f16eda/system/lib/libc/musl/src/thread/pthread_join.c#L45

Reply:
You are right, that was a bad oversight from me. Let me try to fix this.